encoders/firebase-decoders-json/src/main/java/com/google/firebase/decoders/TypeToken.java

```diff
@@ -126,32 +132,26 @@ public String toString() {
    * Plain class types, Generic types, and Wildcard types.
    */
   public static class ClassToken<T> extends TypeToken<T> {
-    private final Class<T> rawType;
     private final TypeTokenContainer typeArguments;
 
     private ClassToken(Class<T> rawType) {
-      this.rawType = rawType;
+      super.rawType = rawType;
       this.typeArguments = TypeTokenContainer.EMPTY;
     }
 
     private ClassToken(Class<T> rawType, TypeTokenContainer typeArguments) {
-      this.rawType = rawType;
+      super.rawType = rawType;
       this.typeArguments = typeArguments;
     }
 
-    @NonNull
-    public Class<T> getRawType() {
-      return rawType;
-    }
-
     @NonNull
     public TypeTokenContainer getTypeArguments() {
       return typeArguments;
     }
 
     @Override
     public int hashCode() {
-      return 11 * rawType.hashCode() + typeArguments.hashCode();
+      return 11 * super.rawType.hashCode() + typeArguments.hashCode();
```

Comment:
I don't think you need `super` here or anywhere above and below.

Reply:
since `ArrayToken` is inner class and `rawType` is private, we need `super` to access the `rawType` or use `getRawType()`

Reply:
how about making it protected instead?

Reply:
Yes, I am thinking if we make it protected, anyone will be able to access it when they extent the `TypeToken`, maybe we loose the advantage of putting `ArrayToken` and `ClassToken` inside `TypeToken`? wdyt?

Reply:
No one will be able to extend it as it has a private constructor